lib/tower_cli/utils/parser.py

```diff
@@ -83,40 +86,28 @@ def string_to_dict(var_string):
     Attempts processing string by 3 different methods in order:
         1. as JSON      2. as YAML      3. as custom key=value syntax
     Throws an error if all of these fail in the standard ways."""
+    # try:
+    #     # Accept all valid "key":value types of json
+    #     return_dict = json.loads(var_string)
+    #     assert type(return_dict) is dict
+    # except (TypeError, AttributeError, ValueError, AssertionError):
     try:
-        # Accept all valid "key":value types of json
-        return_dict = json.loads(var_string)
+        # Accept all JSON and YAML
+        return_dict = yaml.load(var_string)
         assert type(return_dict) is dict
-    except (TypeError, AttributeError, ValueError, AssertionError):
+    except (AttributeError, yaml.YAMLError, AssertionError):
+        # if these fail, parse by key=value syntax
         try:
-            # Accept all valid key: value types of yaml
-            return_dict = yaml.load(var_string)
-            assert type(return_dict) is dict
-        except (AttributeError, yaml.YAMLError, AssertionError):
-            # if these fail, parse by key=value syntax
-            try:
-                return_dict = parse_kv(var_string)
-            except:
-                raise exc.TowerCLIError(
-                    'failed to parse some of the extra '
-                    'variables.\nvariables: \n%s' % var_string
-                )
```

Comment:
Big change: the JSON parser was removed here. That was done to avoid unicode issues. You can't expect to load data with the JSON parser and then dump it from the YAML parser (which you would necessarily do, given the specs of this change), so the only option is to load both types with the YAML parser, and this seems to work fine for all forms of testing with no ill effects.